public/components/FeatureAnywhereContextMenu/AddAlertingMonitor/AddAlertingMonitor.tsx

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useMemo, useState } from 'react';
+import {
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiFlyoutHeader,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiTitle,
+  EuiSpacer,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFormFieldset,
+  EuiCheckableCard,
+} from '@elastic/eui';
+import './styles.scss';
+import CreateNew from './CreateNew';
+import AssociateExisting from './AssociateExisting';
+import _ from 'lodash';
+import { Formik } from 'formik';
+import {
+  getInitialValues,
+  submit,
+} from '../../../pages/CreateMonitor/containers/CreateMonitor/utils/helpers';
+import {
+  createSavedObjectAssociation,
+  validateAssociationIsAllow,
+} from '../../../utils/savedObjectHelper';
+import { SEARCH_TYPE } from '../../../utils/constants';
+import { getClient, getNotifications } from '../../../services';
+
+function AddAlertingMonitor({
+  embeddable,
+  closeFlyout,
+  flyoutMode,
+  setFlyoutMode,
+  monitors,
+  selectedMonitorId,
+  setSelectedMonitorId,
+  index,
+  detectorId,
+  isAssociateAllowed,
+  limitReachedCallout,
+}) {
+  const history = {
+    location: { pathname: '/create-monitor', search: '', hash: '', state: undefined },
+    push: () => null,
+    goBack: closeFlyout,
+  };
+  const setFlyout = () => null;
+  const httpClient = getClient().httpClient;
+  const notifications = getNotifications();
+  const title = embeddable.vis.title;
+  const timeField = embeddable.vis?.data?.aggs?.aggs?.[1]?.params?.field?.displayName;
+  const searchType = flyoutMode === 'adMonitor' ? SEARCH_TYPE.AD : '';
+  const initialValues = useMemo(
+    () =>
+      getInitialValues({ ...history, title, index, timeField, flyoutMode, searchType, detectorId, embeddable }),
+    []
+  );
+  const [isLoading, setIsLoading] = useState(false);
+  const [showErrors, setShowErrors] = useState(false);
+
+  const onCreate = async (values, formikBag) => {
```

Comment:
We should memoize this callback since it is passes down as prop and will cause unnecessary re-render

Reply:
Similarly for other callbacks that are passed down as props

Reply:
I think I got them all